### PR TITLE
gluon-core: preserve interface MAC-address configuration

### PR DESCRIPTION
**Pull request by blocktrron**
This preserve configuration of MAC-addresses for each interface in case OpenWrt sets them using netifd.

This is done by preserving all device-configuration which does not define a bridge or the MAC-address of a bridge.

Closes ##3100

### CODE REVIEW

File: package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces [hunk 1]
```diff
@@ -90,8 +90,15 @@ else
 	sysconfig.single_ifname = lan_ifname or wan_ifname
 end
 
+-- Delete all UCI device esctions of type 'bridge'
```
Comment (skorpy2009) [inline package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces:93]:
```suggestion
-- Delete all UCI device sections of type 'bridge'
```

**Comment by blocktrron** [comment]
Tested on ASUS TUF-AX4200

**Comment by Djfe** [comment]
thank you ❤️

File: package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces [hunk 2]
```diff
@@ -90,8 +90,15 @@ else
...
 end
 
+-- Delete all UCI device sections of type 'bridge'
+-- as well as the ones starting with 'br-'.
+-- Preserve all others to apply MAC address stored in UCI
+uci:foreach('network', 'device',function(s)
+	if s.type == 'bridge' or s.name:match('^br-') then
+		uci:delete('network', s['.name'])
+	end
+end)
```
Comment (neocturne) [inline package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces:100]: Using `delete` in `foreach` has caused weird breakage in the past (including segfaulting Lua at some point). `delete_all` with a function argument must be used instead, see `gluon-respondd/luasrc/lib/gluon/upgrade/400-respondd-firewall`

Comment (blocktrron) [reply package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces:100]: I'll check that

**Pull request merged**